### PR TITLE
Add Kotlin to SharedCode

### PR DESCRIPTION
**Pull request by nachtien**
Adds Kotlin to the SharedCode

### CODE REVIEW

**Review by adelarsq — APPROVED** [review]
Good job!

**Comment by nachtien** [comment]
Looks like clickState is not null.  Updated.  Sorry this took so long!

File: android/sharedCode/src/main/java/com/viro/core/ClickListener.kt
Comment (adelarsq) [reply android/sharedCode/src/main/java/com/viro/core/ClickListener.kt:57]: @nachtien I think that `node` itself **can't** be null. 

> Node can be null

This two makes sense since not every `id` identify a `node` and a `node` can be already removed.

> location can be null

This also make sense since we can pass a "wrong" location.

Would be possible some logging on this cases?

Comment (adelarsq) [reply android/sharedCode/src/main/java/com/viro/core/ClickListener.kt:57]: @nachtien Node isn't nullable. But location is nullable I think.

Comment (nachtien) [reply android/sharedCode/src/main/java/com/viro/core/ClickListener.kt]: Let's just add Kotlin to Gradle for now, I'll convert the other classes first to be sure :)

**Comment by nachtien** [comment]
Note that I bumped Kotlin to `1.6.21`

**Comment by adelarsq** [comment]
@doranteseduardo  @robertjcolley Anything blocked by these changes?

**Comment by robertjcolley** [comment]
I can take a look this weekend. I don't believe anything is blocked by or is blocking this

**Comment by nachtien** [comment]
Let's get this @robertjcolley !

**Comment by hannesa2** [comment]
In my point of view, a community project needs more maintainer with the power to merge.
This micro changes are not merged since January !

**Comment by adelarsq** [comment]
@hannesa2 New contributors are always welcome. Just ping me if interested and we can talk about.
@robertjcolley Thanks for the revision.
@nachtien Thanks for the contribution!

**Pull request merged**
